src/LibHdb++Cassandra.cpp

```
@@ -1119,637 +1286,489 @@ void HdbPPCassandra::extract_and_bind_rw_values(CassStatement* statement,
 	cass_statement_bind_int32(statement, 3, current_time_us);
 
-	future = cass_session_execute(mp_session, statement);
-	cass_future_wait(future);
+	CassError rc = execute_statement(statement);
 
-	CassError rc = CASS_OK;
-	rc = cass_future_error_code(future);
-	if(rc != CASS_OK)
-		print_error(future);
+	if (rc != CASS_OK)
+		throw_execute_exception("ERROR executing insert query", insert_event_str.str(), rc, __func__);
 
-	cass_future_free(future);
-  	cass_statement_free(statement);
-
-	if(rc != CASS_OK)
-	{
-		cerr << __func__ << ": ERROR executing query=" << insert_event_str.str() << endl;
-		return false;
-	}
-
-	return true;
+	TRACE_MSG("Leaving");
 }
 
 
-int HdbPPCassandra::insert_param_Attr(Tango::AttrConfEventData *data, HdbEventDataType ev_data_type)
+void HdbPPCassandra::insert_param_Attr(Tango::AttrConfEventData *data, HdbEventDataType ev_data_type)
 {
-#ifdef _CASS_LIB_DEBUG
-	cout << __func__<< ": entering..." << endl;
-#endif
-    if(data == 0)
-    {
-        cerr << "HdbPPCassandra::insert_param_Attr(): data is a null pointer!" << endl;
-		return -1;
-    }
-    if(data->attr_conf == 0)
-    {
-        cerr << __func__ << ": data->attr_conf is a null pointer!" << endl;
-    }
-    string fqdn_attr_name = data->attr_name;
-    int64_t ev_time = 0;
-    int ev_time_us = 0;
-	try
+	TRACE_MSG("Entering");
+
+	if(data == NULL)
 	{
-		ev_time = ((int64_t) data->get_date().tv_sec) * 1000;
-		ev_time_us = data->get_date().tv_usec;
+		// TODO (SJ) assert this?
+		ERROR_MSG("HdbPPCassandra::insert_param_Attr(): data is a null pointer!" << endl);
+		return ;
 	}
-	catch(Tango::DevFailed &e)
+
+	if(data->attr_conf == NULL)
 	{
-		cerr << "Exception on " << data->attr_name << ":" << endl;
-		for (unsigned int i=0; i<e.errors.length(); i++)
-		{
-			cerr << e.errors[i].reason << endl;
-			cerr << e.errors[i].desc << endl;
-			cerr << e.errors[i].origin << endl;
-		}
-		cerr << endl;
-		return -1;
+		// TODO (SJ) change to return, assert it?
```

The same here: Could you please throw an exception in this case?

---

Also added.